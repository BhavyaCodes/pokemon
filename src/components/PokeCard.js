import React from "react";
import PokemonType from "./PokemonType";

function PokeCard({ pokemon }) {
  const renderTypes = () => {
    if (pokemon.types) {
      return pokemon.types.map((type) => {
        return <PokemonType key={type} type={type} />;
      });
    }
  };

  return (
    <div className="col mb-4">
      <div className="card" tabIndex={0}>
        <img
          src={pokemon.imageUrl}
          className="card-img-top"
          alt={pokemon.name}
        />
        <div className="card-body">
          <h5 className="card-title text-center">{pokemon.name}</h5>
          <p className="card-text text-center">{renderTypes()}</p>
        </div>
      </div>
    </div>
  );
}

export default PokeCard;
