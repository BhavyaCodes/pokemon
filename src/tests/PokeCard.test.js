import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PokeCard from "../components/PokeCard";

it("renders PokeCard component without crashing", () => {
  const mockPokemon = {
    id: 1,
    name: "Mock Pokemon Name",
    imageUrl: "mock-url",
    weight: 1000,
    types: ["mock type 1", "mock type 2"],
  };

  const { asFragment } = render(<PokeCard pokemon={mockPokemon} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders PokeCard component without types if not passed", () => {
  const mockPokemon = {
    id: 1,
    name: "Mock Pokemon Name",
    imageUrl: "mock-url",
    weight: 1000,
  };

  const { asFragment } = render(<PokeCard pokemon={mockPokemon} />);
  expect(asFragment()).toMatchSnapshot();
});
