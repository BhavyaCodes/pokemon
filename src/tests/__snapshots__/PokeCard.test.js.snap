// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokeCard component without crashing 1`] = `
<DocumentFragment>
  <div
    class="col mb-4"
  >
    <div
      class="card"
      tabindex="0"
    >
      <img
        alt="Mock Pokemon Name"
        class="card-img-top"
        src="mock-url"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title text-center"
        >
          Mock Pokemon Name
        </h5>
        <p
          class="card-text text-center"
        >
          <span
            class="badge badge-pill badge-info mx-1"
            data-testid="pokemon-type-mock type 1"
          >
            mock type 1
          </span>
          <span
            class="badge badge-pill badge-info mx-1"
            data-testid="pokemon-type-mock type 2"
          >
            mock type 2
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders PokeCard component without types if not passed 1`] = `
<DocumentFragment>
  <div
    class="col mb-4"
  >
    <div
      class="card"
      tabindex="0"
    >
      <img
        alt="Mock Pokemon Name"
        class="card-img-top"
        src="mock-url"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title text-center"
        >
          Mock Pokemon Name
        </h5>
        <p
          class="card-text text-center"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
